pkg/client/add_events.go

```diff
@@ -184,5 +185,15 @@ func (client *DataSetClient) SendAllAddEventsBuffers() {
 		return true
 	})
 
-	// TODO: wait on all sending to be finished
+	for client.buffersEnqueued.Load() > client.buffersProcessed.Load() {
```

Comment:
So if I understand it correctly, this now makes this `AddEvents` method synchronous / blocking aka it won't return until all the buffers haven't been sent.

That's fine with me, should probably document it in the function doc string though. Is there ever a scenario where we would want a different async version of this function which returns immediately after enqueing buffers?

Reply:
Good point. There has to be some "close" method - https://pkg.go.dev/io#Closer - that will flush everything and waits for it to happen.

I will create new function for this.